grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java

```diff
@@ -305,15 +311,19 @@ private static Map<String, Field> buildFields(Descriptor desc,
                 case BYTE_STRING:
                 case ENUM:
                     // Use field name which is specified in proto file.
-                    builder.put(namePrefix + field.getName(),
-                                new Field(field, parentNames, field.getJavaType()));
+                    builder.put(key, new Field(field, parentNames, field.getJavaType()));
+                    if (useCamelCaseKeys && key != camelCaseKey) {
+                        builder.put(camelCaseKey, new Field(field, parentNames, field.getJavaType()));
+                    }
```

Comment:
Do we need to support both snake_case and camelCase when `useCamelCaseKeys` is true.
If you think it is good to support both of them, how about adding an option for snake case to `HttpJsonTranscodingOptionsBuilder`?

```java
// If both camelCaseQueryParams() and snakeCaseQueryParams() are not set,
// IllegallStateException will be raised by `build()`.
HttpJsonTranscodingOptions.builder()
                          .camelCaseQueryParams(true)
                          .snakeCaseQueryParams(true)
                          .build();
```

The default factory could be created like:
```java
HttpJsonTranscodingOptions DEFAULT = 
  HttpJsonTranscodingOptions.builder()
                            .snakeCaseQueryParams(true)
                            .build();
...

static HttpJsonTranscodingOptions ofDefault() {
     return DEFALT;
}
```